gammapy/catalog/fermi.py

```diff
@@ -660,3 +662,200 @@ def __init__(self, filename=None):
+        ss += '\n'
+
+        # val, err = d['Energy_Flux100'], d['Unc_Energy_Flux100']
+        # ss += 'Energy flux (100 MeV - 100 GeV) : {} +- {} erg cm^-2 s^-1\n'.format(val, err)
+        # ss += 'Detection significance : {}\n'.format(d['Signif_Avg'])
+
+        return ss
+
+    def _get_flux_values(self, prefix='Flux', unit='cm-2 s-1'):
+        if prefix not in ['Flux', 'Unc_Flux']:
+            raise ValueError(
+                "Must be one of the following: 'Flux', 'Unc_Flux'")
+
+        values = [self.data[prefix + _ + 'GeV'] for _ in self._ebounds_suffix]
+        return Quantity(values, unit)
+
+
+    @property
+    def spectral_model(self):
+        """
+        Best fit spectral model `~gammapy.spectrum.models.SpectralModel`.
+        """
+        spec_type = self.data['SpectrumType'].strip()
+        pars = {}
+        pars['amplitude'] = Quantity(
+            self.data['Flux_Density'], 'GeV-1 cm-2 s-1')
+        pars['reference'] = Quantity(self.data['Pivot_Energy'], 'GeV')
+
+        if spec_type == 'PowerLaw':
+            pars['index'] = Quantity(self.data['Spectral_Index'], '')
+            return PowerLaw(**pars)
+
+        elif spec_type == 'LogParabola':
+            pars['alpha'] = Quantity(self.data['Spectral_Index'], '')
+            pars['beta'] = Quantity(self.data['beta'], '')
+            return LogParabola(**pars)
+
+        else:
+            raise ValueError(
+                'Spectral model {} not available'.format(spec_type))
+
+    @property
+    def flux_points(self):
+        """
+        Differential flux points (`~gammapy.spectrum.FluxPoints`).
+        """
+        table = Table()
+        table.meta['SED_TYPE'] = 'flux'
+        e_ref = self._ebounds.log_centers
+        table['e_ref'] = e_ref
+        table['e_min'] = self._ebounds.lower_bounds
+        table['e_max'] = self._ebounds.upper_bounds
+
+        flux = self._get_flux_values()
+        flux_err = self._get_flux_values('Unc_Flux')
+        table['flux'] = flux
+        table['flux_errn'] = np.abs(flux_err[:, 0])
+        table['flux_errp'] = flux_err[:, 1]
+
+        nuFnu = self._get_flux_values('nuFnu', 'erg cm-2 s-1')
+        table['eflux'] = nuFnu
+        table['eflux_errn'] = np.abs(nuFnu * flux_err[:, 0] / flux)
+        table['eflux_errp'] = nuFnu * flux_err[:, 1] / flux
+
+        is_ul = np.isnan(table['flux_errn'])
+        table['is_ul'] = is_ul
+
+        # handle upper limits
+        table['flux_ul'] = np.nan * flux_err.unit
+        table['flux_ul'][is_ul] = table['flux_errp'][is_ul]
+
+        for column in ['flux', 'flux_errp', 'flux_errn']:
+            table[column][is_ul] = np.nan
+
+        # handle upper limits
+        table['eflux_ul'] = np.nan * nuFnu.unit
+        table['eflux_ul'][is_ul] = table['eflux_errp'][is_ul]
+
+        for column in ['eflux', 'eflux_errp', 'eflux_errn']:
+            table[column][is_ul] = np.nan
+
+        # TODO: check if nuFnu is maybe integral flux
```

Comment:
Can you resolve this TODO comment?

In the FITS file I see
```
$ ftlist gll_psch_v11.fit.gz[1] C
HDU 2  

  Col  Name             Format[Units](Range)      Comment
    1 Source_Name        18A                  
    2 RAJ2000            E [deg] (0.000000:360.000) 
    3 DEJ2000            E [deg] (-90.0000:90.0000) 
    4 GLON               E [deg] (0.000000:360.000) 
    5 GLAT               E [deg] (-90.0000:90.0000) 
    6 Conf_95_SemiMajor  E [deg] (0.000000:)  
    7 Conf_95_SemiMinor  E [deg] (0.000000:)  
    8 Conf_95_PosAng     E [deg]              
    9 ROI_num            I                    
   10 Signif_Avg         E (0.000000:)        
   11 Pivot_Energy       E [GeV] (0.000000:)  
   12 Flux_Density       E [photon/cm**2/GeV/s] (0.000000:) 
   13 Unc_Flux_Density   E [photon/cm**2/GeV/s] (0.000000:) 
   14 Flux               E [photon/cm**2/s]   
   15 Unc_Flux           E [photon/cm**2/s]   
   16 Energy_Flux        E [erg/cm**2/s]      
   17 Unc_Energy_Flux    E [erg/cm**2/s]      
   18 Signif_Curve       E                    
   19 SpectrumType       11A                  
   20 Spectral_Index     E (0.000000:10.0000) 
   21 Unc_Spectral_Index E (0.000000:)        
   22 beta               E (-1.00000:1.00000) 
   23 Unc_beta           E (0.000000:)        
   24 PowerLaw_Index     E (0.000000:10.0000) 
   25 Unc_PowerLaw_Index E (0.000000:)        
   26 Flux_Band          5E [photon/cm**2/s]  
   27 Unc_Flux_Band      10E [photon/cm**2/s] 
   28 nuFnu              5E [erg/cm**2/s]     
   29 Sqrt_TS_Band       5E                   
   30 Npred              E (0.000000:)        
   31 HEP_Energy         E [GeV]              
   32 HEP_Prob           E (0.000000:1.00000) 
   33 Variability_BayesBlocks I                    
   34 Extended_Source_Name 18A                  
   35 ASSOC_GAM          18A                  
   36 TEVCAT_FLAG        A                    
   37 ASSOC_TEV          21A                  
   38 CLASS              7A                   
   39 ASSOC1             26A                  
   40 ASSOC2             26A                  
   41 ASSOC_PROB_BAY     E (0.000000:1.00000) 
   42 ASSOC_PROB_LR      E (0.000000:1.00000) 
   43 Redshift           E                    
   44 NuPeak_obs         E [Hz]               
```

So it's clear that `nuFnu` is `E^2 * dN/dE` and not an integral flux, no?

If it's unclear to you, or if the question is really whether we should access that field or other fields, let's figure that out before merging this PR.